addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCBridgeHandler.java

```diff
@@ -0,0 +1,471 @@
+                    buffer[0] = S7.ByteToBCD(calendar.get(Calendar.MONTH) + 1);
+                    buffer[1] = S7.ByteToBCD(calendar.get(Calendar.DATE));
+                }
+
+                int result = client.WriteDBArea(1, bHandler.getAddress(), buffer.length, S7Client.S7WLByte, buffer);
+                if (result != 0) {
+                    logger.error("Can not write data to LOGO!: {}.", S7Client.ErrorText(result));
+                }
+            } else {
+                logger.warn("Can not update not supported channel: {}.", channelUID);
+            }
+        } else {
+            logger.error("Invalid handler {} found.", handler.getClass().getSimpleName());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialize LOGO! bridge handler.");
+
+        final Thing thing = getThing();
+        Objects.requireNonNull(thing, "PLCBridgeHandler: Thing may not be null.");
+
+        synchronized (config) {
+            config = getConfigAs(PLCLogoBridgeConfiguration.class);
+        }
+
+        boolean configured = (config.getAddress() != null);
+        configured = configured && (config.getLocalTSAP() != null);
+        configured = configured && (config.getRemoteTSAP() != null);
+
+        if (configured) {
+            if (client == null) {
+                client = new PLCLogoClient();
+            }
+            configured = connect();
+        }
+
+        if (configured) {
+            final String host = config.getAddress();
+            if (readerJob == null) {
+                final Integer interval = config.getRefreshRate();
+                logger.info("Creating new reader job for {} with interval {} ms.", host, interval);
+                readerJob = scheduler.scheduleWithFixedDelay(dataReader, 100, interval, TimeUnit.MILLISECONDS);
+            }
+            if (rtcJob == null) {
+                logger.info("Creating new RTC job for {} with interval 500 ms.", host);
+                rtcJob = scheduler.scheduleWithFixedDelay(rtcReader, 100, 500, TimeUnit.MILLISECONDS);
+            }
+
+            PLCBridgeHandler.super.initialize();
+        } else {
+            final String message = "Can not initialize LOGO!. Please, check parameter.";
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message);
+            client = null;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Dispose LOGO! bridge handler.");
+        super.dispose();
+
+        if (rtcJob != null) {
+            rtcJob.cancel(false);
+            while (!rtcJob.isDone()) {
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException exception) {
+                    logger.error("Dispose LOGO! bridge handler throw an error: {}.", exception.getMessage());
```

Comment:
This logging does not look really appropriate for an InterruptedException please read what it is about, this no way near an error. Maybe debug logging could apply but you can also just handle it by issue-ing the break.